File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/hac/HacExecutiveAgencyInfoParser.java
package com.shmedo.configlibrary.iot.cmd.parser.hac;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser;
import com.shmedo.configlibrary.iot.model.hac.HacExecutiveAgencyInfo;

import java.util.HashMap;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2022/7/29 <br/>
 * 描述：     TODO
 */
public class HacExecutiveAgencyInfoParser implements IOTResultParser<HacExecutiveAgencyInfo> {
    @Override
    public HacExecutiveAgencyInfo parse(String result) {
        HacExecutiveAgencyInfo info = new HacExecutiveAgencyInfo();
        try {
            String[] keyValues = result.split("&");
            HashMap<String, String> keyValueMap = new HashMap<>();

            for (String keyValue : keyValues) {
                String[] strs = keyValue.split("=");
                if (strs.length < 2) {
                    keyValueMap.put(strs[0], "");
                } else {
                    keyValueMap.put(strs[0], strs[1]);
                }
            }
            info.setDatatype(keyValueMap.getOrDefault("datatype", "NullKey"));
            info.setDatareply(keyValueMap.getOrDefault("datareply", "NullKey"));
            info.setDatainval(keyValueMap.getOrDefault("datainval", "NullKey"));
            info.setCompensatetime(keyValueMap.getOrDefault("compensatetime", "NullKey"));
            info.setDriveaddress(keyValueMap.getOrDefault("driveaddress", "NullKey"));
            info.setDownspeed(keyValueMap.getOrDefault("downspeed", "NullKey"));
            info.setDownwaitetime(keyValueMap.getOrDefault("downwaitetime", "NullKey"));
            info.setUpspeed(keyValueMap.getOrDefault("upspeed", "NullKey"));
            info.setMeaspacing(keyValueMap.getOrDefault("measpacing", "NullKey"));
            info.setMeaintertime(keyValueMap.getOrDefault("meaintertime", "NullKey"));
            info.setInterval_compensation(keyValueMap.getOrDefault("interval_compensation", "NullKey"));
            info.setInterval_fitting(keyValueMap.getOrDefault("interval_fitting", "NullKey"));
            info.setPoint_offset(keyValueMap.getOrDefault("point_offset", "NullKey"));

            return info;
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }
    }

    @Override
    public void validate(String result) {

    }

    @Override
    public IOTCommandType commandType() {
        return IOTCommandType.ADME_HAC_MD_GET_EXECUTIVE_AGENCY;
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/parser/hac/HacExecutiveAgencyInfoParser.kt
package com.shmedo.lib.device.base.iot_cmd.parser.hac

import com.shmedo.lib.device.base.iot_cmd.enums.IOTCommandType
import com.shmedo.lib.device.base.iot_cmd.interfaces.IOTResultParser
import com.shmedo.lib.device.base.iot_cmd.model.hac.HacExecutiveAgencyInfo

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2022/7/29 <br></br>
 * 描述：     TODO
 */
class HacExecutiveAgencyInfoParser : IOTResultParser<HacExecutiveAgencyInfo?> {
    override fun parse(result: String): HacExecutiveAgencyInfo? {
        val info = HacExecutiveAgencyInfo()
        return try {
            val keyValues = result.split("&").toTypedArray()
            val keyValueMap = HashMap<String, String>()
            for (keyValue in keyValues) {
                val strs = keyValue.split("=").toTypedArray()
                if (strs.size < 2) {
                    keyValueMap[strs[0]] = ""
                } else {
                    keyValueMap[strs[0]] = strs[1]
                }
            }
            info.datatype = keyValueMap.getOrDefault("datatype", "NullKey")
            info.datareply = keyValueMap.getOrDefault("datareply", "NullKey")
            info.datainval = keyValueMap.getOrDefault("datainval", "NullKey")
            info.compensatetime = keyValueMap.getOrDefault("compensatetime", "NullKey")
            info.driveaddress = keyValueMap.getOrDefault("driveaddress", "NullKey")
            info.downspeed = keyValueMap.getOrDefault("downspeed", "NullKey")
            info.downwaitetime = keyValueMap.getOrDefault("downwaitetime", "NullKey")
            info.upspeed = keyValueMap.getOrDefault("upspeed", "NullKey")
            info.measpacing = keyValueMap.getOrDefault("measpacing", "NullKey")
            info.meaintertime = keyValueMap.getOrDefault("meaintertime", "NullKey")
            info.interval_compensation =
                keyValueMap.getOrDefault("interval_compensation", "NullKey")
            info.interval_fitting = keyValueMap.getOrDefault("interval_fitting", "NullKey")
            info.point_offset = keyValueMap.getOrDefault("point_offset", "NullKey")
            info
        } catch (ex: Exception) {
            ex.printStackTrace()
            null
        }
    }

    override fun validate(result: String) {}
    override fun commandType(): IOTCommandType {
        return IOTCommandType.ADME_HAC_MD_GET_EXECUTIVE_AGENCY
    }
}